### PR TITLE
Remove code that is redundant with wdlTools, update usages

### PR DESCRIPTION
**Pull request by jdidion**
Also:

* Move a few API calls into DxApi
* Rename a few unclear variables (req, rep)
* Rename SysUtils -> CompressionUtils, since the compression-related functions are the only ones left in dxWDL after moving the rest to wdlTools

### CODE REVIEW

**Comment by jdidion** [comment]
This PR comes second

File: src/test/scala/dx/exec/TaskRunnerTest.scala [hunk 1]
```diff
@@ -127,16 +126,16 @@ class TaskRunnerTest extends AnyFlatSpec with Matchers {
...
         case _: Throwable =>
           None
       }
 
     // Create a clean directory in "/tmp" for the task to use
     val jobHomeDir: Path = Files.createTempDirectory("dxwdl_applet_test")
-    SysUtils.deleteRecursive(jobHomeDir.toFile)
-    SysUtils.safeMkdir(jobHomeDir)
+    Util.deleteRecursive(jobHomeDir)
+    Util.createDirectories(jobHomeDir)
```
Comment (commandlinegirl) [inline src/test/scala/dx/exec/TaskRunnerTest.scala:138]: Not sure in which cases we need these two methods in addition to createTempDirectory() and in which we don't. I made a comment (a question) about it in Pr #429

File: src/main/scala/dx/api/DxApi.scala [hunk 2]
```diff
@@ -431,12 +441,11 @@ case class DxApi(logger: Logger = Logger.Quiet, dxEnv: DXEnvironment = DXEnviron
...
     }
-    val req = JsObject(fields ++ instanceFields ++ dependsFields ++ dwd)
-    logger.traceLimited(s"subjob request=${req.prettyPrint}")
+    val request = fields ++ instanceFields ++ dependsFields ++ dwd
+    logger.traceLimited(s"subjob request=${JsObject(request).prettyPrint}")
 
-    val retval: JsonNode = DXAPI.jobNew(JsUtils.jsonNodeOfJsValue(req), classOf[JsonNode])
-    val info: JsValue = JsUtils.jsValueOfJsonNode(retval)
-    val id: String = info.asJsObject.fields.get("id") match {
+    val info = jobNew(request)
```
[the remark this answers is not in the document]
Comment (jdidion) [reply src/main/scala/dx/api/DxApi.scala:447]: yep - I'll fix these when I find them

File: src/test/scala/dx/exec/TaskRunnerTest.scala (hunk 1 above)
Comment (jdidion) [reply src/test/scala/dx/exec/TaskRunnerTest.scala:138]: right - these are artifacts. I've removed them.

**Pull request merged**